shared_model/interfaces/common_objects/common_objects_factory.hpp

```diff
@@ -13,6 +13,10 @@
 #include "interfaces/common_objects/peer.hpp"
 #include "interfaces/common_objects/types.hpp"
 #include "interfaces/common_objects/domain.hpp"
+#include "interfaces/common_objects/account_asset.hpp"
+#include "interfaces/common_objects/asset.hpp"
+#include "interfaces/common_objects/domain.hpp"
+#include "interfaces/common_objects/signature.hpp"
```

Comment:
Maybe replace it with a forward declaration since it has only declarations?

Reply:
Also domain include seems to be duplication

Reply:
The issues is with signature::SignedType. We cannot forward-declare inner classes.